catalog/migrate/migrate.go

```diff
@@ -0,0 +1,333 @@
+	initialCommitMessage = "repository created"
+)
+
+func (m *Migrate) Close() error {
+	return m.mvccCataloger.Close()
+}
+
+func NewMigrate(db db.Database, cfg *config.Config) (*Migrate, error) {
+	entryCatalog, err := rocks.NewEntryCatalog(cfg, db)
+	if err != nil {
+		return nil, err
+	}
+
+	mvccParams := cfg.GetMvccCatalogerCatalogParams()
+	mvccCataloger := mvcc.NewCataloger(db, mvcc.WithParams(mvccParams))
+	return &Migrate{
+		db:            db,
+		entryCatalog:  entryCatalog,
+		mvccCataloger: mvccCataloger,
+	}, nil
+}
+
+func (m *Migrate) FilterRepository(expr string) error {
+	if expr == "" {
+		m.repositoryRe = nil
+		return nil
+	}
+	repositoryRe, err := regexp.Compile(expr)
+	if err != nil {
+		return err
+	}
+	m.repositoryRe = repositoryRe
+	return nil
+}
+
+func (m *Migrate) Run() error {
+	ctx := context.Background()
+	// logger := logging.FromContext(ctx)
+	fmt.Println("==> Listing repositories to import")
+	repos, err := m.listReposToImport(ctx)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("==> Total", len(repos), "repositories to import")
+	for i, repo := range repos {
+		fmt.Printf("==> Import (%d/%d): %s (storage namespace: %s, default branch: %s)\n", i+1, len(repos), repo.Name, repo.StorageNamespace, repo.DefaultBranch)
+		_, err := m.migrateRepository(ctx, repo)
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			continue
+		}
+	}
+	return nil
+}
+
+func (m *Migrate) listReposToImport(ctx context.Context) ([]*catalog.Repository, error) {
+	var repos []*catalog.Repository
+	var after string
+	for {
+		reposPage, hasMore, err := m.mvccCataloger.ListRepositories(ctx, 100, after)
+		if err != nil {
+			return nil, err
+		}
+		for _, repo := range reposPage {
+			if m.repositoryRe == nil || m.repositoryRe.MatchString(repo.Name) {
+				repos = append(repos, repo)
+			}
+		}
+		if !hasMore || len(reposPage) == 0 {
+			break
+		}
+		after = reposPage[len(reposPage)-1].Name
+	}
+	return repos, nil
+}
+
+func (m *Migrate) migrateRepository(ctx context.Context, repository *catalog.Repository) (*graveler.RepositoryRecord, error) {
+	// get or create repository
+	repo, err := m.getOrCreateTargetRepository(ctx, repository)
+	if err != nil {
+		return nil, err
+	}
+
+	// scan mvcc repository commits by order and reproduce on entry catalog
+	err = m.migrateCommits(ctx, repo)
+	if err != nil {
+		return nil, fmt.Errorf("repository %s: %w", repo.RepositoryID, err)
+	}
+	return repo, nil
+}
+
+func (m *Migrate) getOrCreateTargetRepository(ctx context.Context, repository *catalog.Repository) (*graveler.RepositoryRecord, error) {
+	repoID := graveler.RepositoryID(repository.Name)
+	branchID := graveler.BranchID(repository.DefaultBranch)
+
+	// check and create repository if needed
+	repo, err := m.entryCatalog.GetRepository(ctx, repoID)
+	if errors.Is(err, graveler.ErrRepositoryNotFound) {
+		repo, err = m.entryCatalog.CreateRepository(ctx, repoID, graveler.StorageNamespace(repository.StorageNamespace), branchID)
```

Comment:
Consider just creating the repo and ignoring an "already exists" error.

Reply:
I can do it, but it will modify the behavior a little - when a repository exists I use the information to perform all graveler/entry catalog API - which means that you can create a repository point to a new storage namespace and import.
The import to the new storage namespace.
In this case, it is working before I get the information, if I switch to create - I will need to re-read the repository information after getting 'repository already exists'. WDYT?

Reply:
Yeah, you're right.  Your way is better, my way is SQL on ACID.